### PR TITLE
Fix Rust 1.74 warnings

### PR DESCRIPTION
**Pull request by arpad-m**
Fixes new warnings and clippy changes introduced by version 1.74 of the rust compiler toolchain.

### CODE REVIEW

File: pageserver/src/tenant/timeline.rs [hunk 1]
```diff
@@ -1859,19 +1859,17 @@ impl Timeline {
...
+            res = &mut calculation => { return res }
+            reason = timeline_state_cancellation => {
+                debug!(reason = reason, "cancelling calculation");
+                cancel.cancel();
+                return calculation.await;
+            }
+            reason = taskmgr_shutdown_cancellation => {
+                debug!(reason = reason, "cancelling calculation");
+                cancel.cancel();
+                return calculation.await;
```
Comment (conradludgate) [inline pageserver/src/tenant/timeline.rs:1872]: Can we also remove these returns?

**Review by koivunej — APPROVED** [review]

File: pageserver/src/tenant/timeline.rs (hunk 1 above)
Comment (arpad-m) [reply pageserver/src/tenant/timeline.rs]: we can! i suppose clippy's explicit return lint is turned off in macro expanded code, OR there is something in the macro expansion that makes it not fire.

pushed a commit to remove them.

File: safekeeper/src/timelines_global_map.rs [hunk 2]
```diff
@@ -296,8 +296,8 @@ impl GlobalTimelines {
         global_lock
             .timelines
             .values()
-            .cloned()
             .filter(|t| !t.is_cancelled())
+            .cloned()
```
Comment (petuhovskiy) [inline safekeeper/src/timelines_global_map.rs:300]: FYI the idea here was to `clone` as soon as possible, to release `TIMELINES_STATE` lock sooner. But now I realize that lock is probably still held until the end of the function, so this change LGTM.

**Review by petuhovskiy — APPROVED** [review]

File: safekeeper/src/timelines_global_map.rs (hunk 2 above)
Comment (conradludgate) [reply safekeeper/src/timelines_global_map.rs:300]: Yeah 2 reasons it's held open for the entire duration:
1. cloned() is still lazy on iterators, so it's till referencing the locked contents while collecting
2. temporaries (like mutex guards) remain until the end of the entire statement, and only drop then

Comment (arpad-m) [reply safekeeper/src/timelines_global_map.rs:300]: so to summarize, with the changes the lock is freed sooner even, as less cloning happens, so the whole iterator is done sooner.

**Pull request merged**